src/Common/config_version.h.in

```diff
@@ -2,18 +2,7 @@
 
 // .h autogenerated by cmake!
 
-#cmakedefine01 USE_DBMS_TCP_PROTOCOL_VERSION
-
-#if USE_DBMS_TCP_PROTOCOL_VERSION
-    #include "Core/Defines.h"
-    #ifndef VERSION_REVISION
-        #define VERSION_REVISION DBMS_TCP_PROTOCOL_VERSION
-    #endif
-#else
-    #cmakedefine VERSION_REVISION @VERSION_REVISION@
-#endif
-
-
+#cmakedefine VERSION_REVISION @VERSION_REVISION@
```

Comment:
I though that only one (`DBMS_TCP_PROTOCOL_VERSION`) should be left, what is purpose of the `VERSION_REVISION`?
AFAICS `VERSION_REVISION` after this PR will be only for information (i.e. will contain in logs, system tables, and so on), all checks are performed against `DBMS_TCP_PROTOCOL_VERSION`, but should it be that way? That said that why not use `DBMS_TCP_PROTOCOL_VERSION` for this? (since this is the real protocol version)

Reply:
VERSION_REVISION is just some number for server version, not related to TCP protocol, like VERSION_MAJOR and so on. It duplicates the other version numbers, but a single number might be more convenient to use in some cases, so I'd keep it.

Reply:
Then maybe it worth to add a comment that `VERSION_REVISION` has nothing common with `DBMS_TCP_PROTOCOL_VERSION`?

Reply:
No objections to that. People are still getting confused all the time.

Reply:
#25771 25771